.github/workflows/code-analysis.yml

```diff
@@ -0,0 +1,37 @@
+name: Code Analysis
+
+on:
+  push:
+    branches: [ master, android-v2 ]
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+
+jobs:
+  sonar:
+    name: SonarScanner
+    runs-on: ubuntu-20.04
+    permissions:
+      pull-requests: read
+      actions: read
+      contents: read
+      security-events: write
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Build scanner image
+        run: |
+          make scanner/image
+      - name: Build app and scan code
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          SONAR_OUT_DIR: sonar_build_wrapper_output
+          PR_NUMBER: ${{ github.event.number }}
+          PR_BRANCH: ${{ github.HEAD_REF }}
+          PR_BASE: ${{ github.BASE_REF }}
+          CONNECT_TTY_TO_DOCKER: 0
+          WITH_TFLITE: 1
+          WITH_PIXEL: 1
+        run: |
+          make docker/scan
```

Comment:
Aren't there some limits of how much CPU time we can use in our github actions?
Can we run our of them if we use github to build our C++ code?

Reply:
The GitHub action is free for open-source repo. Certainly, there is some limitation or quota but I couldn't find it on the internet.